File: src/main/java/finance/repositories/AccountRepository.java
package finance.repositories;

import finance.model.Account;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface AccountRepository extends JpaRepository<Account, Long> {
    @Query(value = "SELECT  * FROM t_account WHERE account_type='credit' AND active_status = 'Y' AND account_Name_Owner != 'grand.total_dummy' AND account_Name_Owner !='credit.total_dummy' AND account_Name_Owner != 'debit.total_dummy' ORDER BY account_Name_Owner", nativeQuery = true)
    List<Account> findActiveAccounts();
}

File: src/main/kotlin/finance/repositories/AccountRepository.kt
package finance.repositories

import finance.models.Account
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface AccountRepository<T : Account> : JpaRepository<T, Long> {
    fun findByAccountId(transactionId: Long?): Account
    fun findByAccountNameOwner(accountNameOwner: String): Optional<Account>
    fun findByActiveStatusOrderByAccountNameOwner(activeStatus: String): List<Account>
}
